Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -122,15 +135,105 @@ public class OptimizelyUserContext {
                              userId: userId,
                              attributes: attributes,
                              eventTags: eventTags)
     }
+    
+}
 
+// MARK: - ForcedDecisions
+
+extension OptimizelyUserContext {
+    
+    public func setForcedDecision(flagKey: String,
+                                  ruleKey: String? = nil,
+                                  variationKey: String) -> Bool {
+        
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = savedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            savedDecisions[index].variationKey = variationKey
+        } else {
+            savedDecisions.append(SavedDecision(flagKey: flagKey, ruleKey: ruleKey, variationKey: variationKey))
+        }
+        
+        return true
+    }
+    
+    public func getForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return nil
+        }
+        
+        return findForcedDecision(flagKey: flagKey, ruleKey: ruleKey)
+    }
+    
+    public func removeForcedDecision(flagKey: String, ruleKey: String? = nil) -> Bool {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = savedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            _ = savedDecisions.remove(at: index)
+            return true
+        }
+        
+        return false
+    }
+    
+    public func removeForcedDecisions() -> Bool {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        savedDecisions.removeAll()
+        return true
+    }
+    
+    func findForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
+        guard savedDecisions.count > 0 else { return nil }
+        
+        if let item = savedDecisions.filter({ $0.flagKey == flagKey && $0.ruleKey == ruleKey }).first {
+            return item.variationKey
+        }
+        
+        return nil
+    }
+    
+    func findValidatedForcedDecision(flagKey: String,
+                                     ruleKey: String?,
+                                     options: [OptimizelyDecideOption]? = nil) -> DecisionResponse<Variation> {
+        let reasons = DecisionReasons(options: options)
+        
+        if let variationKey = findForcedDecision(flagKey: flagKey, ruleKey: ruleKey) {
+            if let variation = optimizely?.getFlagVariationByKey(flagKey: flagKey, variationKey: variationKey) {
```

Comment:
in the `else` of this and above one should have `debug` reasons.

Reply:
I refactored to add the reasons for this else. "Else" one for the above is intentionally dropped, since all decisions are with no forced decisions by default, polluting debug messages.